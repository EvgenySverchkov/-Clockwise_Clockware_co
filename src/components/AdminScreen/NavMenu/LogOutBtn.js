import React from "react";
import { useDispatch } from "react-redux";
import { Link } from "react-router-dom";
import { toogleAuth } from "../../../store/adminPanel/services/actions";

const LogOutBtn = () => {
  const dispatch = useDispatch();

  function handler() {
    sessionStorage.removeItem("token");
    dispatch(toogleAuth(false));
  }
  return (
    <div className="mr-1">
      <Link
        to="/admin"
        onClick={handler}
        className="btn btn-secondary"
        aria-haspopup="true"
        aria-expanded="false"
      >
        LogOut
      </Link>
    </div>
  );
};

export default LogOutBtn;
