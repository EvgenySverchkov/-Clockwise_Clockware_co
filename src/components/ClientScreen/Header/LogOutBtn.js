import React from "react";
import { Link } from "react-router-dom";

function LogOutBtn({ handler }) {
  return (
    <li className="nav-item">
      <Link className="nav-link" to="/client" onClick={handler}>
        Logout
      </Link>
    </li>
  );
}

export default LogOutBtn;
