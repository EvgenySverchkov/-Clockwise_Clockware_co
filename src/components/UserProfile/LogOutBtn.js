import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { Link } from "react-router-dom";

import { addCurrentOrderToState } from "../../store/ordersManagement/actions";
import { toogleAuthClient } from "../../store/auth/actions";

function LogOutBtn() {
  const state = useSelector((state) => {
    return { currentOrder: state.ordersReducer.currentOrder };
  });
  const dispatch = useDispatch();

  function logOutHandl() {
    ["user", "token"].forEach((item) => localStorage.removeItem(item));
    dispatch(addCurrentOrderToState({ ...state.currentOrder, email: "" }));
    dispatch(toogleAuthClient(false));
  }
  return (
    <Link className="btn btn-danger" to="/" onClick={logOutHandl}>
      Logout
    </Link>
  );
}

export default LogOutBtn;
