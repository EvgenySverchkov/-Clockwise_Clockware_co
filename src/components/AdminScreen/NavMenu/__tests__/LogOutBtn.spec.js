import React from "react";
import {create} from "react-test-renderer";
import {Provider} from "react-redux";
import configMockStore from "redux-mock-store";
import {MemoryRouter} from "react-router-dom"

import LogOutBtn from "../LogOutBtn";

const mockStore = configMockStore()

it("Test of <LogOutBtn /> on admin side", ()=>{
    const store = mockStore({})
    const component = create(
        <Provider store = {store}>
            <MemoryRouter initialEntries={["/"]}>
                <LogOutBtn />
            </MemoryRouter>
        </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
});