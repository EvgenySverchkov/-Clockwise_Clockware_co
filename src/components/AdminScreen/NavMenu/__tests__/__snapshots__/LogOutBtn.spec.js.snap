// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <LogOutBtn /> on admin side 1`] = `
<div
  className="mr-1"
>
  <a
    aria-expanded="false"
    aria-haspopup="true"
    className="btn btn-danger"
    href="/admin"
    onClick={[Function]}
  >
    LogOut
  </a>
</div>
`;
